src/main/java/org/ehrbase/fhirbridge/ehr/converter/PatientCompositionConverter.java

```diff
@@ -0,0 +1,128 @@
+
+    @Override
+    public Patient fromComposition(GECCOPersonendatenComposition composition) {
+        // TODO: Implement
+        return null;
+    }
+
+    @Override
+    public GECCOPersonendatenComposition toComposition(Patient fhir_patient) {
+        if (fhir_patient == null) {
+            return null;
+        }
+
+        //create composition and contained archetype objects
+        GECCOPersonendatenComposition composition = new GECCOPersonendatenComposition();
+        PersonendatenAdminEntry person_data = new PersonendatenAdminEntry();
+
+        // don't set feeder audit - plans to generalize it exist (mapping doesn't need to deal with it and currently if throws a null pointer exception in mapping test)
+        //FeederAudit fa = CommonData.constructFeederAudit(fhir_patient);
+        //composition.setFeederAudit(fa);
+
+        /* NOTE There are more fields included in the FHIR resource (but not marked as "support mandatory")
+        which are also considered in the template. But the example does only include age, date of birth and ethnic group.*/
+
+        //assemble composition
+        composition.setAlter(get_age_from_fhir(fhir_patient));
+        person_data.setDatenZurGeburt(get_data_on_birth(fhir_patient));
+        person_data.setEthnischerHintergrund(get_ethnic_background_data(fhir_patient));
+        person_data.setSubject(new PartySelf());
+        person_data.setLanguage(Language.DE);
+
+        composition.setPersonendaten(person_data);
+
+        //NOTE not sure if this is the best value for composition startTime since it refers to the documentation of the age, but its the only info on time of documentation...
+        composition.setStartTimeValue(composition.getAlter().getTimeValue());
+
+        // Required fields by API
+        composition.setLanguage(Language.DE);
+        composition.setLocation("test"); //FIXME: sensible value
+        composition.setSettingDefiningCode(Setting.SECONDARY_MEDICAL_CARE);
+        composition.setTerritory(Territory.DE);
+        composition.setCategoryDefiningCode(Category.EVENT);
+        composition.setComposer(new PartySelf()); //FIXME: sensible value
+
+        return composition;
+    }
+
+    private List<EthnischerHintergrundCluster> get_ethnic_background_data(Patient fhir_patient) {
+        List<EthnischerHintergrundCluster> items = new ArrayList<>();
+        try {
+            Extension extension_ethnic_group = fhir_patient.getExtensionByUrl("https://www.netzwerk-universitaetsmedizin.de/fhir/StructureDefinition/ethnic-group");
+            Coding ethnic_group = (Coding) extension_ethnic_group.getValue();
+            EthnischerHintergrundCluster ec = new EthnischerHintergrundCluster();
+            ec.setEthnischerHintergrundDefiningCode(EthnischerHintergrundDefiningCode.get_by_SNOMED_code(ethnic_group.getCode()));
+            items.add(ec);
+        } catch (Exception e) {
```

Comment:
Do you really want to catch ANY Exception here

Reply:
...yes, I guess you don't like that? I have no clue how a more elegant solution could look like. I added a hint on what happened since tests revealed I did not get any sensible message before.